### PR TITLE
chore: minimum required roles guard api-v2

### PR DESCRIPTION
**Pull request by ThyMinimalDev**
## What does this PR do?

Roles guard, following minimum required role to access a resource

- USER SYSADMIN can access everything
- TEAM OWNER can access TEAM ADMIN resources and TEAM MEMBER resources
- TEAM ADMIN can access TEAM MEMBER resources but not TEAM OWNER resources
- ORG OWNER can access ORG ADMIN resources and ORG MEMBER resources
- ORG ADMIN can access ORG MEMBER resources but not ORG OWNER resources
- ORG ADMIN and ORG OWNER can access TEAM OWNER/ADMIN/MEMBER resources 
- If a resource is for users with the role USER, no need to use this guard

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [x] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

### CODE REVIEW

File: apps/api/v2/src/lib/roles/constants.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  `ORG_${MembershipRole.OWNER}`,
+  `ORG_${MembershipRole.ADMIN}`,
+  `ORG_${MembershipRole.MEMBER}`,
+] as const;
+export const TEAM_ROLES = [
+  `TEAM_${MembershipRole.OWNER}`,
+  `TEAM_${MembershipRole.ADMIN}`,
+  `TEAM_${MembershipRole.MEMBER}`,
+] as const;
+export const USER_ROLES = [`USER_${UserPermissionRole.ADMIN}`, `USER_${UserPermissionRole.USER}`] as const;
```
Comment (alishaz-polymath) [inline apps/api/v2/src/lib/roles/constants.ts:14]: What is the purpose of this again? 🤔

Comment (alishaz-polymath) [reply apps/api/v2/src/lib/roles/constants.ts:14]: `USER_ADMIN` is `SYSADMIN` no? Unless I am missing something 🤔

Comment (ThyMinimalDev) [reply apps/api/v2/src/lib/roles/constants.ts:14]: oh yes correct 🤦  ill make the change

**Review by alishaz-polymath — DISMISSED** [review]
LGTM 🚀 🚀

**Review by alishaz-polymath — APPROVED** [review]

**Comment by supalarry** [comment]
Looked at this with fresh mind again:

We now have legacy [membership-roles.decorator.ts](https://github.com/calcom/cal.com/pull/15576/files#diff-c7ddccd438f3bf5d485fba344114bf3945e1a82838a308a2cf083330ea1c8161) with [organization-roles.guard.ts](https://github.com/calcom/cal.com/pull/15576/files#diff-e8d2c7f0be80ca4c67a2af4cec5c591055c1e9081753d3d2381fb49a5cb951ef) invoking `MembershipsRepository.findOrgUserMembership`. This was prior to having "organizations" logic.


The new stuff is [roles.decorator.ts](https://github.com/calcom/cal.com/pull/15576/files#diff-c7d42aafef930b16a24c1ef550ef4dacde704b76cf1b1e991cc9afd0d67efcaa) with [roles.guard.ts](https://github.com/calcom/cal.com/pull/15576/files#diff-fc87e5bc7d3f66198c396dbc19bf45dfb20f8a8a4765e5a2389a8165216b890c) invoking `MembershipsRepository.findMembershipByTeamId`.

The new roles guard designed with orgs in mind is called "roles.guard.ts" but previous one without orgs in mind "organizations-roles.guard.ts". Also, `MembershipsRepository.findOrgUserMembership` and `MembershipsRepository.findMembershipByTeamId` are basically identical.

The older `OrganizationRolesGuard` guard + `MebershipRoles` decorator are used in billing and oauth-clients controllers, so can we re-use the new guard + decorator there and delete them and `MembershipsRepository.findMembershipByTeamId` in repository? We ought to I think.

**Pull request merged**